File: frontend-mealsonwheels/src/types/foodSafety/IngredientDTO.ts
// src/types/management/IngredientDTO.ts

export interface IngredientDTO {
  id: number;
  name: string;
  expirationDate: string; // ISO date string (e.g., "2025-06-06")
  quantity: number;
  unit: string;
  storageConditions: string;
}

File: backend-mealsonwheels/src/main/java/com/merrymeal/mealsonwheels/dto copy/IngredientDTO.java
package com.merrymeal.mealsonwheels_backend.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.*;

import java.time.LocalDate;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class IngredientDTO {

    private Long id;

    @NotBlank(message = "Ingredient name is required")
    @Size(max = 255)
    private String name;

    private LocalDate expirationDate;

    private Double quantity;

    @Size(max = 255)
    private String unit;

    @Size(max = 255)
    private String storageConditions;
}
